File: 41-50/Q50.cpp
//Q50.cpp
class Solution {
public:
    double myPow(double x, int n) {
        long long p=n;
        if(p<0) x=1/x, p=-p;
        double r=1;
        while(p){
            if(p&1) r*=x;
            x*=x;
            p>>=1;
        }
        return r;
    }
};

File: 41-50/Q50.py
#Q50.py
class Solution:
    def myPow(self, x, n):
        if n<0:
            x=1/x; n=-n
        r=1.0
        while n:
            if n&1: r*=x
            x*=x
            n>>=1
        return r
